backend/api/views.py

```diff
@@ -1,77 +1,74 @@
+class DeleteUser(APIView):
+    permission_classes = (IsAuthenticated,)
 
+    def delete(self, request):
+        usermail = request.user.username
+        user = FiboUser.objects.get(email=usermail)
+        user.delete()
+        return Response(f'Your user account was deleted', status=status.HTTP_200_OK)
 
-@api_view(['POST'])
-def registerUser(request):
-    # Note that username and email are the same
-    user = FiboUser.objects.create_user(username=request.data['email'], email=request.data['email'],
-                                        password=request.data['password'])
-    return Response(f'User with id {user.id} and email {user.email} was created', status=status.HTTP_201_CREATED)
 
+class RegisterUser(APIView):
+    permission_classes = (AllowAny, )
 
-@api_view(['POST'])
-def authenticateUser(request):
-    # Note that username and email are the same
-    user = authenticate(username=request.data['email'], password=request.data['password'])
-    if user is not None:
-        # TODO login(request,user)
-        # redirect to a success page
-        return Response(f'Authenticated user {user.email}', status=status.HTTP_200_OK)
-    else:
-        return Response('Invalid login', status=status.HTTP_401_UNAUTHORIZED)
+    def post(self, request):
+            # Note that username and email are the same
+            user = FiboUser.objects.create_user(username=request.data['email'], email=request.data['email'],
+                                                password=request.data['password'])
+            return Response(f'User with id {user.id} and email {user.email} was created', status=status.HTTP_201_CREATED)
 
 
-@api_view(['GET'])
-def getRoutes(request):
-    routes = [
-        {
-            'Endpoint': '/users/register',
-            'method': 'POST',
-            'body': {'email': '', 'password': ''},
-            'description': 'Register Endpoint for a new user!'
-        },
-        {
-            'Endpoint': '/users/authenticate',
-            'method': 'POST',
-            'body': {'email': '', 'password': ''},
-            'description': 'Authenticates user with given email and password'
-        },
-        {
-            'Endpoint': '/users/<id>/delete',
-            'method': 'DELETE',
-            'body': None,
-            'description': 'Deletes user with given id'
-        },
-        {
-            'Endpoint': '/users/<id>',
-            'method': 'GET',
-            'body': None,
-            'description': 'Returns user with given id'
-        },
-        {
-            'Endpoint': '/users/<id>/update',
-            'method': 'PUT',
-            'body': {'email': '', 'newPassword': '', 'oldPassword': ''},
-            'description': 'Not implemented yet. Updates user with data sent in put request'
-        },
-    ]
-    return Response(routes)
+class GetRoutes(APIView):
+    permission_classes = (IsAuthenticated, )
+
+    def get(self):
+        routes = [
```

Comment:
```suggestion
    def get(self,_):
```

we need to add the pass the request to the get method in order to work because the method will automatically always be called with 2 parameters

Reply:
done